sliceop/f64s/f64s.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2019 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Common operations on float64 slices
+package f64s
+
+
+// Return a slice with all elements for which f(x) returns true
+func Filter(dst, src []float64, f func(v float64) bool) []float64 {
+	if dst == nil {
+		dst = []float64{}
+	}
+	for _, x := range src {
+		if f(x) {
+			dst = append(dst, x)
+		}
+	}
+	return dst	
+}
+
+// Return a slice of {f(x_i)} when src = {x_i}
+func Map(dst, src []float64, f func(v float64) float64) []float64 {
+	if dst == nil {
+		dst = []float64{}
+	}
+	for _, x := range src {
+		dst = append(dst, f(x))
+	}
+	return dst
+}
+
+// Return a slice of all indices corresponding to elements for which f(x) is true
+func Find(dst []int, src []float64, f func(v float64) bool) []int {
+	if dst == nil {
+		dst = []int{}
+	}
+	for i, x := range src {
```

Comment:
and here I'd add:
```go
dst = dst[:0]
```
(with a matching comment in the doc-string of `Find`.)

Reply:
OK I am not sure to understand the logic of this one ... After some googling, I understand this is the clean the slice. But then, shouldn't we do this for the three functions (and *before* the loop)?

Reply:
yes, `dst = dst[:0]` will modify `dst` to have `len(dst)==0` while still keeping its underlying capacity unchanged.

and, yes, it should be put just before the loop.
we use it only for `Filter` and `Find`, because we don't know the final number of elements in `dst`.

but for `Map`, we know that `len(dst)==len(src)`, so we don't have to use `append` (just random access via `dst[i]`)